tensorflow/BUILD

```diff
@@ -939,7 +939,6 @@ tf_cc_shared_object(
         "//tensorflow/core/common_runtime:core_cpu_impl",
         "//tensorflow/core:framework_internal_impl",
         "//tensorflow/core/common_runtime/gpu:gpu_runtime_impl",
-        "//tensorflow/core/common_runtime/pluggable_device:pluggable_device_runtime_impl",
```

Comment:
No, it will cause link failure if remove this unless we also remove `if_static` in core/common_runtime/pluggable_device/BUILD:115, as Kulin did, however, if we remove this `if_static`, MacOs has duplicate symbol issue as we met before.

```
    -] + if_static([
        # Temporary workaround for duplicated symbols issues.
        ":pluggable_device_runtime_impl",
        -"//tensorflow/core/common_runtime:bfc_allocator",
        -"//tensorflow/core/common_runtime:local_device",
        -"//tensorflow/core/common_runtime:process_state",
```

 link error:
_pywrap_tensorflow_internal.so: undefined symbol: _ZN10tensorflow19PluggableDeviceUtil18DeviceToDeviceCopyEPNS_13DeviceContextES2_PNS_6DeviceES4_NS_19AllocatorAttributesES5_PKNS_6TensorEPS6_iSt8functionIFvRKNS_6StatusEEE

Reply:
Thanks @penpornk and @jzhoulon for the review. 

That's correct, `if_static` was causing the plugin registry to not be shared across the so and the dylib for libtensorflow_framework on MacOS. I tried running couple of tests which were failing earlier and couldn't reproduce them locally. I will try to add back the libs which we need for headers, but having the static initialization will cause for plugins on MacOS.

Reply:
@penpornk can you please take a look if the changes make sense ? I was looking at the failure logs, and I couldn't reproduce these locally. The unit tests mentioned below such as:
//tensorflow/examples/adding_an_op:fact_test   
are working fine.

https://tensorflow-ci.intel.com/job/tensorflow-mkl-linux-cpu-pr/9345/artifact/oob_unittest_all_failures.log